great_expectations/experimental/datasource/fabric.py

```diff
@@ -0,0 +1,394 @@
+        filters: Optional[Dict[str, str]] = None,
+        fully_qualified_columns: Optional[bool] = None,
+        num_rows: Optional[int] = None,
+        pandas_convert_dtypes: bool = True,
+        use_xmla: bool = False,
+    ) -> PowerBIMeasure:
+        """Adds a PowerBIMeasure asset to this datasource.
+
+        Args:
+            name: The name of this asset.
+            order_by: A list of Sorters or Sorter strings.
+            batch_metadata: BatchMetadata we want to associate with this DataAsset and all batches derived from it.
+
+        Returns:
+            The asset that is added to the datasource.
+        """
+        order_by_sorters: list[Sorter] = self.parse_order_by_sorters(order_by=order_by)
+        asset = PowerBIMeasure(
+            name=name,
+            order_by=order_by_sorters,
+            batch_metadata=batch_metadata or {},
+            groupby_columns=groupby_columns,
+            measure=measure,
+            # TODO: require custom serde for keys that are tuples
+            filters=filters,
+            fully_qualified_columns=fully_qualified_columns,
+            num_rows=num_rows,
+            pandas_convert_dtypes=pandas_convert_dtypes,
+            use_xmla=use_xmla,
+        )
+        return self._add_asset(asset)
+
+    def add_powerbi_table_asset(  # noqa: PLR0913
+        self,
+        name: str,
+        table: str,
+        order_by: Optional[SortersDefinition] = None,
+        batch_metadata: Optional[BatchMetadata] = None,
+        fully_qualified_columns: bool = False,
+        num_rows: Optional[int] = None,
+        multiindex_hierarchies: bool = False,
+        pandas_convert_dtypes: bool = True,
+    ) -> PowerBITable:
+        """Adds a PowerBITable asset to this datasource.
+
+        Args:
+            name: The name of this table asset.
+            table_name: The table where the data resides.
+            schema: The schema that holds the table.
+            order_by: A list of Sorters or Sorter strings.
+            batch_metadata: BatchMetadata we want to associate with this DataAsset and all batches derived from it.
+
+        Returns:
+            The asset that is added to the datasource.
+        """
+        order_by_sorters: list[Sorter] = self.parse_order_by_sorters(order_by=order_by)
+        asset = PowerBITable(
+            name=name,
+            order_by=order_by_sorters,
+            batch_metadata=batch_metadata or {},
+            table=table,
+            fully_qualified_columns=fully_qualified_columns,
+            num_rows=num_rows,
+            multiindex_hierarchies=multiindex_hierarchies,
+            pandas_convert_dtypes=pandas_convert_dtypes,
+        )
+        return self._add_asset(asset)
+
+    @staticmethod
+    def _running_on_fabric() -> bool:
+        if (
+            os.environ.get("AZURE_SERVICE", None)  # noqa: TID251 # needed for fabric
+            != "Microsoft.ProjectArcadia"
+        ):
```

Comment:
I thought we were avoiding `os.environ` usage outside of config files - is there somewhere else we could grab this from?

Reply:
Nit - the `None` is unnecessary and I'd love to have "Microsoft.ProjectArcadia" be a constant (perhaps at the file or class level)